zokrates_core/src/flatten/mod.rs

```diff
@@ -30,12 +30,16 @@ pub struct Flattener {
     bijection: BiMap<String, FlatVariable>,
 }
 impl Flattener {
+    pub fn flatten<T: Field>(p: TypedProg<T>) -> FlatProg<T> {
+        Flattener::new(T::get_required_bits()).flatten_program(p)
```

Comment:
Is this really a better design? Doesn't it assume that the flatten function is only called once?

Reply:
It does. If we want to reuse Flattener instances we would need to reset their state. I feel like creating a new one each time is cleaner and cheap.

Reply:
Okay!